procsimulator.py

```diff
@@ -1361,6 +1398,43 @@ def execInstr(self):
                 for flag in workingFlags:
                     self.flags[flag] = workingFlags[flag]
 
+        elif t == InstrType.multiplylong:
+            op1 = self.regs[misc['operandsmul'][0]].get()
+            op2 = self.regs[misc['operandsmul'][1]].get()
+            destHi = misc['rdHi']
+            destLo = misc['rdLo']
+
+            res = (self.regs[destHi].get() << 32) + self.regs[destLo].get()
+            if misc['signed']:
+                if op1 & 1 << 31:
+                    op1 -= 1 << 32
+                if op2 & 1 << 31:
+                    op2 -= 1 << 32
+                if res & 1 << 63:
+                    res -= 1 << 64
+
+            if misc['accumulate']:
+                # MLAL
+                res += (op1 * op2)
+            else:
+                # MULL
+                res = op1 * op2
+
+            if misc['signed'] and res < 0:
+                res += 1 << 64
+
+            self.regs[destHi].set(res >> 32 & 0xFFFFFFFF)
+            self.regs[destLo].set(res & 0xFFFFFFFF)
+
+            # Z are set, V and C is set to "meaningless value" (see ARM spec 4.7.2), V is unaffected
```

Comment:
The right section is 4.8.2, and V is indeed affected (contrary to MUL).

Reply:
You are right for the section, but it say "Both the C and V flags are set to meaningless values".

Reply:
Yes, the behavior of the code is the right one (actually, any behavior would be fine since anything can be a meaningless value!), I was just pointing out that the comment was contradictory, since it ends with "V is unaffected"